cli/src/main/java/hudson/cli/CLI.java

```diff
@@ -405,6 +424,9 @@ public void upgrade() {
     public static void main(final String[] _args) throws Exception {
         try {
             System.exit(_main(_args));
+        } catch (NotTalkingToJenkinsException ex) {
+            System.err.println(ex.getMessage());
+            System.exit(-1);
```

Comment:
Right, looking at the table in https://issues.jenkins-ci.org/browse/JENKINS-32273, it seems like `3`.